iommu/boot/runtest.sh

```diff
@@ -0,0 +1,164 @@
+
+	    if [ ${code} -ne 0 ]; then
+		echo "Fail: error changing boot loader." |
+		tee -a "${OUTPUTFILE}"
+		report_result "${TEST}/boot_loader" "FAIL" 0
+	    else
+		echo "${line}" > $CurrentBootOptions
+		echo "Reboot now!" | tee -a "${OUTPUTFILE}"
+		report_result "${TEST}/boot_loader" "PASS" 0
+		rhts-reboot
+	    fi
+	else
+            # The reboot has finished. Verify the cmdline.
+	    echo "New cmdline: $(cat /proc/cmdline)" | tee -a "${OUTPUTFILE}"
+
+            grep "$(cat $CurrentBootOptions)" /proc/cmdline
+	    code=$?
+	    # remove spaces for reporting boot option to beaker
+	    CurrentBootOptionsReport=$(cat $CurrentBootOptions | sed 's/\ /-/')	    
+
+	    if [ ${code} -ne 0 ]; then
+		echo "Fail: error booting kernel with specified cmdline" |
+		tee -a "${OUTPUTFILE}"
+
+		report_result "${TEST}/$CurrentBootOptionsReport" "FAIL" 0
+		rm $CurrentBootOptions
+	        /sbin/grubby --remove-args="${line}" \
+		 --update-kernel="${default}"
+        	sed -i "/$line\$/d" $bootOptionsFile
+	    else
+       		echo "boot options persisted through reboot." | tee -a "${OUTPUTFILE}"
+		report_result "${TEST}/$CurrentBootOptionsReport" "PASS" 0
+	        rm $CurrentBootOptions
+                /sbin/grubby --remove-args="${line}" \
+                 --update-kernel="${default}"
+        	sed -i "/$line\$/d" $bootOptionsFile
+	    fi
+	fi
+
+
+
+    done < $bootOptionsFile
+}
+
+function dmesgErrors() {
+    dmesgLineNumber=0
+    
+    # find any iommu errors in dmesg/messages file
+    while read -r dmesgLine; do
+	dmesgLineNumber=$(($dmesgLineNumber+1))
+	journalctl | grep "$dmesgLine"
+	    code=$?
+	    if [ ${code} -ne 1 ]; then
+		echo "Fail: the following iommu regex matched in dmesg:" |
+		tee -a "${OUTPUTFILE}"
+		echo "$dmesgLine" | tee -a "${OUTPUTFILE}"
+		echo "see TESTOUT.log for actual message or $dmesgReport for report" |
+		tee -a "${OUTPUTFILE}"
+		echo "$dmesgLineNumber FAIL $dmesgLine" >> $dmesgReport
+	    else
+		echo "$dmesgLineNumber PASS $dmesgLine" >> $dmesgReport
+	    fi
+    done < $dmesgErrors
+
+    # report pass/fail to beaker if errors were found, upload report
+    grep FAIL $dmesgReport
+    dmesgReportCode=$?
+
+    if [ ${dmesgReportCode} -ne 1 ]; then
+	report_result "${TEST}/iommu-dmesg" "FAIL" 0
+    else
+	report_result "${TEST}/iommu-dmesg" "PASS" 0
+    fi
+    
+    rhts-submit-log -l $dmesgReport
+
+}
+
+function cleanupTest() {
+    rm $dmesgReport
+}
+
+if [[ -n $CMDLINEARGS ]]; then
+    if [ -z "${REBOOTCOUNT}" ] || [ "${REBOOTCOUNT}" -eq 0 ]; then
+       IFS=':'
+       for i in $CMDLINEARGS; do
+	   echo $i >> $CustomBootOptions
+       done
+    fi
+    bootOptions $CustomBootOptions
+    dmesgErrors
+else
+    if [[ $cpuvendor = "GenuineIntel" ]]; then
+	bootOptions $DefaultBootOptionsIntel
+	dmesgErrors
+    elif [[ $cpuvendor = "AuthenticAMD" ]]; then
+       	bootOptions $DefaultBootOptionsAMD
+	dmesgErrors
+    else
+	report_result "${TEST}/nonAMDorIntelProcessor" "FAIL" 0
```

Comment:
Should this be SKIP instead, as it's not the patch developer's fault that the test is running on unsupported machine?

Reply:
Sounds good. That was in the original test, and I didn't even know `SKIP` was a thing in this harness :sweat_smile: 

Which reminds me: how much are we supposed to review/change the tests when copying them?